routes/index.js

```diff
@@ -442,14 +442,20 @@ router.get('/getEvents', function(req, res) {
     collection.find({}, function(e,docs){
       var temp = new Array;
       docs.forEach(function(val){
-        if(req.session.usr.role == "EVI" && val.toEVI == "true") {
-          temp.push(val);
-        }
-        else if (req.session.usr.role == "innostaja" && val.toInnostaja == "true") {
-          temp.push(val);
-        }
-        else if (req.session.usr.role == "admin") {
-          temp.push(val);
+        if(req.session.usr.role == "EVI" && val.toEVI == "true"
+        || req.session.usr.role == "innostaja" && val.toInnostaja == "true"
+        || req.session.usr.role == "admin") {
+          if (!val.releaseTime || req.session.usr.role == "admin") { // has no release time
```

Comment:
For data integrity I think all event objects should have releaseDate. (new Date()) This should instead check if that time has already passed.